app/models/post.rb

```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  def update_post_counter
+    self.comment_counter = comments.count
+    save
+  end
```

Comment:
- [ ] According to project requirement, a method to update post_count should be available in the post model, currently your method is update post comment count not post count, kindly revamp your code , it should look similar to this. however since you have called this method in your comment model you can create another method for update post count, and name them properly
```
def update_post_counter
  author.increment!(:posts_counter)
end
```